### DISCUSSION TITLE
Boolean Expression Evaluations

### DISCUSSION THREAD

**Discussion by fakalit** [ethanblake4/dart_eval#34] [Q&A] [answered]
First of all, congrats on the amazing and ambitious project. 

I've been playing around with it, and I'm not sure how to approach simple boolean evaluations like this:
Following expressions all throw the matching exceptions.

**eval("true")**
```
E/flutter (29117): [ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: CompileError: Unknown literal type BooleanLiteralImpl
E/flutter (29117): #0      parseLiteral (package:dart_eval/src/eval/compiler/expression/literal.dart:31:3)

```
**eval("2 > 1")**
```
E/flutter (29117): [ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: CompileError: Cannot box F-1:bool extends Object
E/flutter (29117): #0      Variable.boxIfNeeded (package:dart_eval/src/eval/compiler/variable.dart:69:7)
```

**eval("3 > 2 && 2 > 5")**
```
E/flutter (29117): [ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: CompileError: Unknown binary operator AMPERSAND_AMPERSAND
E/flutter (29117): #0      compileBinaryExpression (package:dart_eval/src/eval/compiler/expression/binary.dart:27:43)

```

Thanks.

**Comment by ethanblake4** [comment, marked as answer]
Hi @fakalit !

Thanks for the report. I've created issue #38, #39, and #40 to cover these cases and I'll fix them shortly.

For the second issue (2 > 1) you should be able to get around it for now by doing `eval("bool main() { return 2 > 1; }")`. By specifying the function's return type (instead of letting it to default to `dynamic`) the compiler will leave the return value unboxed which should allow this to succeed. Of course, that should not be required and I will fix it.

**Comment by ethanblake4** [comment]
Fixed in latest master. Will look to get a new version out on Pub soon

**Comment by fakalit** [comment]
Thanks for the very quick response and fix @ethanblake4 ! Hope this library takes off, it certainly deserves it.

**Comment by ethanblake4** [comment]
@fakalit New version 0.4 on Pub is out with these fixes
